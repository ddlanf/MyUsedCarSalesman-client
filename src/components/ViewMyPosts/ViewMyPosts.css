.view-my-posts-post-box{
  margin-top: 12%;
  margin-bottom: 5%;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: flex-start;
  align-content: space-around;
  flex-wrap: wrap;
}

.view-my-posts-create-post{
  text-align: center;
  margin-top: 10%;
}

.create-post-button{
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 10%;
  min-width: 100px;
  margin-top: 30px;
  height: 5vw;
  min-height: 30px;
  font-size: 15px;
  margin-bottom: 8%;
  background-color: #dfe32e;
  border-radius: 10%;
  border: 1px solid #3c484d;
  font-weight: bold;
}

.create-post-button:hover{
  background-color: #996060;
  transition: 0.5s;
}

.view-my-posts-link{
  color: black;
  text-decoration: none;
}


@media screen and (min-width: 1000px) {
  .view-my-posts-post-box{
      margin-top: 6%;
  }

  .create-post-button{
    font-size: 30px;
    height: 100px;
    width: 200px; 
  }
}