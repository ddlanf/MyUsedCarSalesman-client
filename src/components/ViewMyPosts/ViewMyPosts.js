import React, { Component } from 'react'
import './ViewMyPosts.css'
import { Link } from 'react-router-dom'
import MyPosts from './MyPosts/MyPosts'

export default class ViewMyPosts extends Component {
    render() {
        return (
            <>
                <section className="view-posts-post-box">
                    <MyPosts/>
                </section>
                <button className="create-post-button"> 
                    <Link
                        className="view-my-post-link"
                        to="./make-post"
                        >
                        Create Post
                    </Link>
                </button>
            </>
        )
    }
}
