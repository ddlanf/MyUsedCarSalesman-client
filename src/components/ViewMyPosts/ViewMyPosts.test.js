import React from 'react';
import ReactDOM from 'react-dom';
import ViewMyPosts from './ViewMyPosts';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('ViewMyPosts component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                        <ViewMyPosts/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                    <ViewMyPosts/>
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});