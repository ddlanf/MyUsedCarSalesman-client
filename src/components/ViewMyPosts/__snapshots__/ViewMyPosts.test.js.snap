// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewMyPosts component renders the UI as expected 1`] = `
Array [
  <a
    className="view-my-posts-link"
    href="/make-post"
    onClick={[Function]}
  >
    <button
      className="create-post-button"
    >
      Create Post
    </button>
  </a>,
  <section
    className="view-my-posts-post-box"
  />,
]
`;
